tests/nosetests/pyanaconda_tests/module_subscription_tests.py

```diff
@@ -897,6 +897,29 @@ def insights_property_test(self):
           False
         )
 
+    def registered_property_test(self):
+        """Test the IsRegistered property."""
+        # should be false by default
+        self.assertFalse(self.subscription_interface.IsRegistered)
+
+        # this property can't be set by client as it is set as the result of
+        # subscription attempts, so we need to call the internal module interface
+        # via a custom setter
+
+        def custom_setter(value):
+            self.subscription_module.set_registered(value)
+
+        # check the property is True and the signal was emitted
+        # - we use fake setter as there is no public setter
+        self._check_dbus_property(
+          "IsRegistered",
+          True,
+          setter=custom_setter
```

Comment:
Couldn't you just add here ``self.subscription_module.set_registered`` directly?

Reply:
AFAIK we want to use `_check_dbus_property()` to test properties where possible & all the other properties with private setters are tested like this.

Reply:
I know. I don't mean to not use _check_dbus_property. I was talking about creating your custom setter which only call set_registered with the same value. So you can skip creating the custom_setter and just use the set_registered directly instead.

I should have point this to the custom setter to make it more obvious.